src/language-markdown/printer-markdown.js

```diff
@@ -451,7 +451,7 @@ function alignListPrefix(prefix, options) {
   return (
     prefix +
     " ".repeat(
-      additionalSpaces >= 4 ? 0 : additionalSpaces // 4+ will cause indented code block
+      additionalSpaces >= 2 ? 0 : additionalSpaces // 4+ will cause indented code block
```

Comment:
This does not look like the correct fix. As mentioned in the comment, the `4` here is for avoiding causing unexpected indented code block.

The fix should be in https://github.com/prettier/prettier/blob/b861448dbf4f5fac83ac9555adf87989f03ee3a5/src/language-markdown/preprocess.js#L192-L199

where we have to identify if an unordered list is aligned. 

Few test cases:

```md
<!-- maybe aligned (treated as not aligned) -->
- 123
- 123

<!-- aligned (similar to ordered list, an easy way to mark it as aligned) -->
-  123
- 123

<!-- not aligned -->
- 123
-  123

<!-- maybe aligned + aligned = aligned -->
- 123
- 123
  1.  123
  1. 123

<!-- maybe aligned + maybe aligned = maybe aligned (treated as not aligned) -->
- 123
- 123
  1. 123
  1. 123

<!-- maybe aligned + maybe aligned = maybe aligned (treated as not aligned) -->
- 123
- 123
  - 123
  - 123

<!-- maybe aligned + aligned = aligned -->
- 123
- 123
  -  123
  - 123
```

Reply:
Please explain the following:

The following is maybe aligned:

```
- 123
- 123
```

While the following is aligned:

```
-  123
- 123
```

Reply:
What I meant **algined** is that it should be formatted as aligned (i.e., add spaces to align `tabWidth`). In the first case, it's the minimal requirement for a list item so we don't know if it should be aligned. For the second case, it's just a trick to mark it as aligned.